README.md

```diff
@@ -625,9 +625,37 @@ For example:
   </appender>
 ```
 
-The async appenders will never block the logging thread.
+#### RingBuffer Full
+
+The async appenders will by default never block the logging thread.
 If the RingBuffer is full (e.g. due to slow network, etc), then events will be dropped.
 
+Alternatively, you can configure the appender to wait until space becomes available instead of dropping the events immediately. This may come in handy when you want to rely on the buffering and the async nature of the appender but don't want to loose any event in case of large logging bursts that exceed the size of the RingBuffer.
+
+The behaviour of the appender when the RingBuffer is controlled by the `appendTimeout` configuration property:
+
+| `appendTimeout` | Behaviour when RingBuffer is full                                      |
+|-----------------|------------------------------------------------------------------------|
+| `< 0`           | disable timeout and wait until space is available                      |
+| `0`             | no timeout, give up immediately and drop event (this is the *default*) |
+| `> 0`           | retry during the specified amount of time                              |
+
+
+Logging threads waiting for space in the RingBuffer wake up periodically at a frequency defined by  `appendRetryFrequency` (default `50ms`). You may increase this frequency for faster reaction time at the expense of higher CPU usage.
+
+When the appender drops an event, it emits a warning status message every `droppedWarnFrequency` consecutive dropped events. Another status message is emitted when the drop period is over and a first event is succesfully enqueued reporting the total number of events that were dropped.
+
+
+#### Graceful Shutdown
+
+When stopped, async appenders perform a _graceful shutdown_ waiting until all events in the buffer are processed and the buffer is empty.
+The maximum time to wait is configured by the `shutdownGracePeriod` parameter and is set to `1 minute` by default.
+
+Events still in the buffer after this period is elapsed are dropped and the appender is stopped.
```

Comment:
It would be beneficial to mention how to gracefully stop logback here.

There's already a sentence about cleanly shutting down logback a bit further down in the Async Appenders section.  Could just move that sentence up to this sub-section.

> In order to guarantees that logged messages have had a chance to be processed by asynchronous appenders (including the TCP appender) and ensure background threads have been stopped, you'll need to [cleanly shut down logback](http://logback.qos.ch/manual/configuration.html#stopContext) when your application exits.